pkg/apis/core/v1/defaults.go

```diff
@@ -125,7 +125,8 @@ func SetDefaults_Service(obj *v1.Service) {
 	// Defaults ExternalTrafficPolicy field for NodePort / LoadBalancer service
 	// to Global for consistency.
 	if (obj.Spec.Type == v1.ServiceTypeNodePort ||
-		obj.Spec.Type == v1.ServiceTypeLoadBalancer) &&
+		obj.Spec.Type == v1.ServiceTypeLoadBalancer ||
```

Comment:
I think this wants to be `NodePort || LB || (ClusterIP && externalIPs)`, doesn't it?  We don't want to support eTP on `type=ExternalName`

Reply:
externalIPs being set when type=ExternalName is a bug.  Proxies skip it.